src/main/kotlin/no/nav/familie/tilbake/kravgrunnlag/task/BehandleStatusmeldingTask.kt

```diff
@@ -0,0 +1,30 @@
+package no.nav.familie.tilbake.kravgrunnlag.task
+
+import no.nav.familie.prosessering.AsyncTaskStep
+import no.nav.familie.prosessering.TaskStepBeskrivelse
+import no.nav.familie.prosessering.domene.Task
+import no.nav.familie.tilbake.kravgrunnlag.KravvedtakstatusService
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+
+@Service
+@TaskStepBeskrivelse(taskStepType = BehandleStatusmeldingTask.BEHANDLE_STATUSMELDING,
+                     maxAntallFeil = 3,
+                     beskrivelse = "Håndter mottatt statusmelding fra oppdrag",
+                     triggerTidVedFeilISekunder = 60 * 5)
+class BehandleStatusmeldingTask(private val kravvedtakstatusService: KravvedtakstatusService) : AsyncTaskStep {
+
+    private val log = LoggerFactory.getLogger(this::class.java)
+    private val secureLog = LoggerFactory.getLogger("secureLogger")
+
+    override fun doTask(task: Task) {
+        log.info("BehandleStatusmeldingTask prosesserer med id=${task.id} og metadata ${task.metadata}")
+        secureLog.info("BehandleStatusmeldingTask prosesserer med id=${task.id} og metadata ${task.metadata}")
+        kravvedtakstatusService.håndterMottattStatusmelding(task.payload)
+    }
+
+    companion object {
+
+        const val BEHANDLE_STATUSMELDING = "behandleStatusmelding"
```

Comment:
Jeg vet at dette er sånn disse konstantene navngis veldig mage andre steder, men jeg synes ikke "NAVNET_PÅ_TASKEN" gir så veldig mye verdi som navn. Det blir bare veldig mye å lese uten å gi noe mer verdi. Jeg Synes TYPE er bedre. Da får man taskStepType = BehandleStatusmeldingTask.TYPE og tilsvarende for opprettelse av domeneobjektene. Synes det er greiere å lese.

Reply:
fikset